Add dataset ID column to datasets list

dbczumar opened the pull request:
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/dbczumar/mlflow/pull/18984?quickstart=1)

#### Install mlflow from this PR

```
# mlflow
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/18984/merge
# mlflow-skinny
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/18984/merge#subdirectory=libs/skinny
```

For Databricks, use the following command:

```
%sh curl -LsSf https://raw.githubusercontent.com/mlflow/mlflow/HEAD/dev/install-skinny.sh | sh -s pull/18984/merge
```

</p>
</details>

<!-- Remove unused checkboxes except for the patch release section at the bottom -->

### Related Issues/PRs

<!-- Uncomment 'Resolve' if this PR can close the linked items. -->
<!-- Resolve --> #xxx

### What changes are proposed in this pull request?

Add dataset ID column to datasets list

**Motivation:** MLflow does not allow you to load a dataset by name using the MLflow SDK, unless you're using Databricks. You must pass the dataset ID to the SDK. **However, the MLflow UI doesn't display the dataset ID, which makes this workflow pretty painful** (need to do a search request for all datasets via the SDK or look at network tab...).

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

<!-- Attach code, screenshot, video used for manual testing here. -->

### Does this PR require documentation update?

- [X] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Release Notes

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

<!-- Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change. -->

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/prompts`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/projects`: MLproject format, project running backends
- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages

<!--
Insert an empty named anchor here to allow jumping to this section with a fragment URL
(e.g. https://github.com/mlflow/mlflow/pull/123#user-content-release-note-category).
Note that GitHub prefixes anchor names in markdown with "user-content-".
-->

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

<!-- Do not modify or remove any text inside the parentheses. Keep both checkboxes below. -->

- [X] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar commented on mlflow/server/js/src/experiment-tracking/pages/experiment-evaluation-datasets/components/ExperimentEvaluationDatasetsListTable.tsx:
```diff
@@ -37,10 +39,22 @@ const COLUMN_IDS = {
+    id: COLUMN_IDS.DATASET_ID,
+    accessorKey: 'dataset_id',
+    header: 'Dataset ID',
+    enableSorting: false,
```
No need to support sorting on dataset ID...

dbczumar committed:
fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

daniellok-db commented on mlflow/server/js/src/experiment-tracking/pages/experiment-evaluation-datasets/components/ExperimentEvaluationDatasetsIdCell.tsx:
```diff
@@ -0,0 +1,43 @@
+
+  return (
+    <Tooltip
+      content={showTooltip ? 'Copied!' : 'Click to copy'}
```
nit: i18n (we don't i18n in OSS but it's a best practice so we don't have i18n issues on backport) also probably unnecessary to have this be a ternary since if the tooltip is not shown they won't see the message anyway

```suggestion
      content={
        <FormattedMessage defaultMessage="Copied!" description="Tooltip text shown when the user copies the tag text" />
      }
```

daniellok-db commented on mlflow/server/js/src/experiment-tracking/pages/experiment-evaluation-datasets/components/ExperimentEvaluationDatasetsIdCell.tsx:
```diff
@@ -0,0 +1,43 @@
+  return (
+    <Tooltip
+      content={showTooltip ? 'Copied!' : 'Click to copy'}
+      open={showTooltip ? true : undefined}
```
nit: does it need to be undefined, or is it fine if it's `false`? 
```suggestion
      open={showTooltip}
```

daniellok-db replied:
oh i see, we want to dynamically change it so that when the user hovers, it shows "Click to copy", and when they click it shows "Copied!" and stays open? if that's the case then ignore this and the above comment

daniellok-db reviewed (approved):
lgtm!

dbczumar committed:
Make copy tooltip behavior consistent with MLflow standard pattern

- Use useClipboard hook instead of navigator.clipboard directly
- Add proper tooltip state management with open/onOpenChange
- Implement tooltipOpen = open || copied to keep tooltip visible for 3 seconds
- Add proper timer cleanup with useRef for timeout tracking
- Remove ternary for open prop - now properly controlled with boolean

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
Use useCopyController hook instead of reimplementing copy logic

Reuse MLflow's standard useCopyController hook for consistent copy behavior.
This is much cleaner and maintains consistency with other copy buttons in the UI.

Signed-off-by: dbczumar <corey.zumar@databricks.com>

dbczumar committed:
Fix prettier formatting for DatasetIdCell

Signed-off-by: dbczumar <corey.zumar@databricks.com>
